ktor-utils/common/src/io/ktor/util/Lock.kt

```diff
@@ -6,14 +6,22 @@
 
 package io.ktor.util
 
+import kotlinx.io.core.*
+
 @InternalAPI
-expect class Lock() {
+expect class Lock() : Closeable {
     fun lock()
     fun unlock()
 }
 
 @InternalAPI
-inline fun <R> Lock.use(block: () -> R): R {
+expect class ReadWriteLock() : Closeable {
+    fun readLock(): LockTicket
```

Comment:
These methods would be just fine, but Java's RWL has the same with different behaviour. So I'd prefer to name it differently to focus attention on the fact that it acquires the lock.